### PR TITLE
Add request estimate feature

### PR DESCRIPTION
**Pull request by chrisenytc**
Add some improvements and features
- Products
- Apply Surge
- Apply Cars availability
- Request estimate
- Request cancellation
- Travis support
- Rspec task
- More guides

### CODE REVIEW

File: lib/uber/api/products.rb [hunk 1]
```diff
@@ -9,6 +9,14 @@ def products(*args)
         arguments = Uber::Arguments.new(args)
         perform_with_objects(:get, "/v1/products", arguments.options, Product)
       end
+
```
Comment (sishen) [inline lib/uber/api/products.rb:12]: @chrisenytc can you resubmit the pull request by replacing tab with "2 space", thank you!

Comment (chrisenytc) [inline lib/uber/api/products.rb:12]: @sishen sure

**Comment by chrisenytc** [comment]
Done @sishen

**Comment by chrisenytc** [comment]
@sishen can you approve the pull request and update the gem to the rubygems?

**Comment by sishen** [comment]
@chrisenytc Sorry for the late. Had issues to access the Gmail in China due to GFW.

**Comment by chrisenytc** [comment]
No problem @sishen, thank for the merge.

**Pull request merged**